### PR TITLE
`wasm-mutate`: Add a mutator for custom sections

### PR DESCRIPTION
**Pull request by fitzgen**

### CODE REVIEW

File: crates/wasmparser/src/parser.rs [hunk 1]
```diff
@@ -538,27 +527,13 @@ impl Parser {
...
-                        data_offset: content.original_position(),
-                        data: content.remaining_buffer(),
-                        range,
-                    });
-                }
+                if id == 0 {}
 
                 match (self.encoding, id) {
                     // Module sections
+                    (Encoding::Module, 0) => {
```
Comment (alexcrichton) [inline crates/wasmparser/src/parser.rs:534]: I think this needs to match either encoding, right? (both components and core wasm modules)

**Review by alexcrichton — APPROVED** [review]
Looks good to me! Can `CustomSectionReader` have private fields though field with accessors?

File: crates/wasmparser/src/parser.rs (hunk 1 above)
Comment (fitzgen) [reply crates/wasmparser/src/parser.rs]: Done.

**Comment by fitzgen** [comment]
> Can CustomSectionReader have private fields though field with accessors?

Done. Made them `pub(crate)` tho so that I didn't have to update all the `assert_matches!` parse tests.

**Pull request merged**